src/services/quandl-service.js

```diff
@@ -12,6 +12,31 @@
         QUANDL_URL = 'https://www.quandl.com/api/v3/',
         QUANDL_WIKI = 'datasets/WIKI/';
 
+    var useApiKey = true;
+
+    class Observable {
+        constructor() {
+            this.registry = {};
+        }
+
+        on(eventName, fn, scope) {
+            if (!this.registry[eventName]) {
+                this.registry[eventName] = [];
+            }
+
+            this.registry[eventName].push({
+                fn: fn,
+                scope: scope
+            });
+        }
+
+        notify(eventName, value) {
+            this.registry[eventName].forEach((entry) => {
+                entry.fn.call(entry.scope, value);
```

Comment:
Forcing the use of lambdas can make it awkward to reuse the functions sent anywhere else.
It'll also makes it really verbose on the subscribers side if we add more events.

I'm happy to change it if you feel strongly about it though :)